Module/src/PSWordCloud/NewWordCloudCommand.cs

```diff
@@ -475,11 +475,8 @@ protected override void BeginProcessing()
         /// </summary>
         protected override void ProcessRecord()
         {
-            var text = MyInvocation.ExpectingInput
-                    ? new[] { InputObject.BaseObject as string }
-                    : InputObject.BaseObject as string[] ?? new[] { InputObject.BaseObject as string };
-
-            _wordProcessingTasks = _wordProcessingTasks ?? new List<Task<IEnumerable<string>>>(text.Length);
+            IEnumerable<string> text = NormalizeInput(InputObject);
+            _wordProcessingTasks = _wordProcessingTasks ?? new List<Task<IEnumerable<string>>>(text.Count());
```

Comment:
`Count` is going to make `NormalizeInput` invoke twice essentially.  Go with a best guess estimate for capacity or take the hit in resize time if most input is going to have a small count.

Maybe have a separate method that quickly analyzes the `InputObject` to guess how many object it contains like:

```csharp
int GetEstimatedCapacity(PSObject input)
{
    if (input.BaseObject is string stringValue)
    {
        return 1;
    }

    if (input.BaseObject is IList list) return list.Count;
    // etc
}
```